src/comm/Connection.cc

```diff
@@ -125,8 +127,39 @@ void Connection::set_max_buffer_size(uint32_t max_buffer_size)
     _max_buffer_size = std::min(MAX_BUFFER_SIZE, _max_buffer_size);
 }
 
+const std::string& Connection::get_source() { return _source; }
+
+short Connection::get_source_family() { return _source_family; }
+
+const std::string& Connection::get_encryption() { return _encryption; }
+
+std::string Connection::source_family_name(short source_family)
+{
+    switch (source_family) {
+        case AF_INET:
+            return "ipv4";
+        case AF_UNSPEC:
+            return "unspec";
+        default:
+            return "unk";
+    }
+}
+
 ConnMetrics::~ConnMetrics() = default;
 
 void ConnMetrics::observe_bytes_sent(std::size_t /*bytes_sent*/) {}
 
 void ConnMetrics::observe_bytes_recv(std::size_t /*bytes_recv*/) {}
+
+class TestConnection : public Connection
+{
+   public:
+    TestConnection() : Connection() { _source = "testing"; }
+    size_t read(uint8_t* /*buffer*/, size_t /*length*/) override { return 0; }
+    size_t write(const uint8_t* /*buffer*/, size_t /*length*/) override { return 0; }
+};
+
+std::shared_ptr< Connection > Connection::testingConnection()
+{
+    return std::make_shared< TestConnection >();
+}
```

Comment:
Does this really need to be here?

Reply:
it doesn't need to be HERE, but what needs to happen is in athena the test framework needs to create a connection to feed into the testing framework, so this was the first logical place to put it.